integration_tests/juniper_tests/src/codegen/interface_attr.rs

```diff
@@ -652,49 +499,25 @@ mod trivial_async {
         async fn id(&self) -> &str;
```

Comment:
@tyranron the trouble with current approach is implementation that non-`async` trait method can be implemented with non-`async` object method only, while `async` trait method can be implemented with both `async` and non-`async` object methods. This happens because non-`async` methods are represented with both `Field` and `AsyncField` traits, while `async` ones with `AsyncField` only. I should definitely add pretty assertion for it. 

So implementing non-`async` trait method with `async` object field will error.

```rust
#[graphql_interface]
trait Character {
    fn id(&self) -> String;
}

impl Human {
    // Errors
    async fn id(&seld) -> String {
        /// ...
    }
}
```

Reply:
I don't think this is possible to get rid of asyncness in traits, as they are represented with `enum`s, which are implementing `Field` and/or `AsyncField` reflection traits.

Reply:
Also looking at test case

```rust
#[graphql_object(scalar = S: ScalarValue, impl = CharacterValue<S>)]
impl Human {
    async fn id<'a, S: ScalarValue>(&self, executor: &'a Executor<'_, '_, (), S>) -> &'a str {
        executor.look_ahead().field_name()
    }

    fn home_planet(&self) -> &str {
        &self.home_planet
    }

    async fn info<'b>(&'b self, _arg: Option<i32>) -> &'b str {
        &self.home_planet
    }
}
```

I guess we can transform this into 

```rust
#[graphql_object(scalar = S, impl = CharacterValue<S>)]
//     No `: ScalarValue` ^
impl Human {
    async fn id<'a, S: ScalarValue>(&self, executor: &'a Executor<'_, '_, (), S>) -> &'a str {
        executor.look_ahead().field_name()
    }

    fn home_planet(&self) -> &str {
        &self.home_planet
    }

    async fn info<'b>(&'b self, _arg: Option<i32>) -> &'b str {
        &self.home_planet
    }
}
```

by searching for `S` generic not only in `impl`, but also in object methods

Reply:
@ilslv 

> I guess we can transform this into

Not the case for this PR. Once we land it, we will reconsider all the attributes in a separate PR anyway.

But I do tend to have a visual separation between types and type params for library user.

> the trouble with current approach is implementation that non-`async` trait method can be implemented with non-`async` object method only, while `async` trait method can be implemented with both `async` and non-`async` object methods. This happens because non-`async` methods are represented with both `Field` and `AsyncField` traits, while `async` ones with `AsyncField` only. I should definitely add pretty assertion for it.

Well I don't think that deciding such things should be a library user burden. Ideally, we don't want sync/async coloring to be a thing here. We do want working it transparantely, providing a meaningful compilation error if something goes wrong.

So... as a GraphQL interface is just a bunch of fields, which are checked structuraly by signatures, the asyncness notion doesn't depend on the interface itself, it rather depends on how the _objects implementing the this interface__ are resolved. So our job here is to have pretty const assertions and that's it.